lib/charms/tempo_k8s/v1/charm_tracing.py

```diff
@@ -127,11 +132,29 @@ def get_current_span() -> Union[Span, None]:
     return cast(Span, span)
 
 
+def _get_tracer_from_context(ctx: Context) -> Optional[ContextVar]:
+    tracers = [v for v in ctx if v is not None and v.name == "tracer"]
```

Comment:
Just curious:
- Where is the `"tracer"` literal coming from?
- How could we end up with multiple contextvars that have the same name?
- Are we assuming here that different "tracer" contextvars would have the same value? I.e. why do we return `tracersp[0]` only?

Reply:
Tracer was defined here: https://github.com/canonical/tempo-k8s-operator/blob/switch-to-otlp-http/lib/charms/tempo_k8s/v1/charm_tracing.py#L100

The fallback code comes from the (unexplained yet) realization that the same tracer is available under a different reference in the context of a function decorated with `@trace`. We created #60 to investigate further.